src/Npgsql/PostgresEnvironment.cs

```diff
@@ -55,4 +84,95 @@ static class PostgresEnvironment
         => GetHomeDir() is string homedir
             ? Path.Combine(homedir, RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "postgresql" : ".postgresql")
             : null;
-}
\ No newline at end of file
+
+    static string? GetSystemPostgresDir() => Environment.GetEnvironmentVariable("PGSYSCONFDIR");
+
+    static string? GetDefaultSystemPostgresDir()
+    {
+        try
+        {
+            return ExecuteCommandAsync("pg_config", async: false, "--sysconfdir").GetAwaiter().GetResult()?.FirstOrDefault();
+        }
+        catch
+        {
+            return null;
+        }
+    }
+
+    internal static ValueTask<string[]?> ExecuteCommandAsync(
```

Comment:
We do not in general add `Async` suffixes to methods unless they're public (the reason being, whether they're async or not depends on `async` flag).

Reply:
Fixed.